diff -up ./CMakeLists.txt.orig ./CMakeLists.txt
--- ./CMakeLists.txt.orig	2018-08-03 20:12:49.982125501 +0300
+++ ./CMakeLists.txt	2018-08-03 20:21:10.367587136 +0300
@@ -78,7 +78,7 @@ cmake_pop_check_state()
 
 find_package(Threads)
 
-find_package(Zlib)
+find_package(ZLIB)
 find_package(Antlr3 REQUIRED)
 cmake_push_check_state(RESET)
 set(CMAKE_REQUIRED_INCLUDES ${ANTLR3C_INCLUDE_DIRS})
@@ -127,7 +127,7 @@ set(LIBS_PRIVATE "-L${antlr3c_library_pa
 get_filename_component(bctoolbox_library_path "${BCTOOLBOX_CORE_LIBRARIES}" PATH)
 set(LIBS_PRIVATE "${LIBS_PRIVATE} -L${bctoolbox_library_path} -lbctoolbox")
 if(ZLIB_FOUND)
-	set(REQUIRES_PRIVATE "${REQUIRES_PRIVATE} z")
+	set(LIBS_PRIVATE "${LIBS_PRIVATE} -lz")
 endif()
 if(HAVE_LIBDL)
 	set(LIBS_PRIVATE "${LIBS_PRIVATE} -ldl")

